selectinf/randomized/approx_reference.py

```diff
@@ -37,27 +38,39 @@ def __init__(self,
 
         self.solve_args = solve_args
 
-        result, inverse_info = query.selective_MLE(observed_target,
-                                                   cov_target,
-                                                   cov_target_score,
-                                                   solve_args=solve_args)[:2]
+        linear_part = query.sampler.affine_con.linear_part
+        offset = query.sampler.affine_con.offset
 
-        self.linear_part = query.sampler.affine_con.linear_part
-        self.offset = query.sampler.affine_con.offset
+        opt_linear = query.opt_linear
 
-        self.regress_opt = query.sampler.logdens_transform[0]
-        self.cond_mean = query.cond_mean
-        self.prec_opt = np.linalg.inv(query.cond_cov)
-        self.cond_cov = query.cond_cov
+        observed_score = query.observed_score_state + query.observed_subgrad
 
-        self.observed_target = observed_target
-        self.cov_target_score = cov_target_score
+        result, inverse_info, log_ref = query.selective_MLE(observed_target,
+                                                            cov_target,
+                                                            regress_target_score,
+                                                            dispersion)
+
+        cond_cov = query.cond_cov
+        self.cond_precision = np.linalg.inv(cond_cov)
+        self.cond_cov = cond_cov
         self.cov_target = cov_target
+        self.prec_target = np.linalg.inv(cov_target)
+
+        self.observed_target = observed_target
+        self.regress_target_score = regress_target_score
+        self.opt_linear = opt_linear
+        self.observed_score = observed_score
+
+        self.M1 = query.M1 * dispersion
```

Comment:
It seems everytime we see M1,M2,M3 they get modified this way by dispersion. I think this can be done when first computing them...